tests/integration/api/controllers/well-known.spec.js

```diff
@@ -0,0 +1,43 @@
+const chai = require('chai');
+const utils = require('@utils');
+
+const unauthenticatedGetRequestOptions = {
+  method: 'GET',
+  noAuth: true
+};
+
+describe('well-known', () => {
+  describe('GET /.well-known/assetlinks.json', () => {
+    afterEach(() => utils.revertSettings(true));
+
+    it('returns 404 if not configured', async () => {
+      await utils.request({
+        path: '/.well-known/assetlinks.json',
+        ...unauthenticatedGetRequestOptions,
+      })
+        .then(() => chai.expect.fail('should have thrown'))
+        .catch(error => {
+          chai.expect(error.response.statusCode).to.equal(404);
+        });
```

Comment:
I see this is your entry into the [age-old debate](https://github.com/medic/cht-core/pull/9016#discussion_r1573096160) of how to expect async errors in mocha...  😅  Though, does this `catch` not also cover the `expect.fail`?

(Not looking for any changes here.  Just wish there was a more clean way to do this in general....)

Reply:
😅 the syntax came naturally when typing this out but I agree the error message when `chai.expect.fail()` gets executed sucks
```
  1) well-known
       GET /.well-known/assetlinks.json
         returns 404 if not configured:
     TypeError: Cannot read properties of undefined (reading 'statusCode')
      at /home/mokhtar/app/medic/cht-core/tests/integration/api/controllers/well-known.spec.js:30:38
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Context.<anonymous> (/home/mokhtar/app/medic/cht-core/tests/integration/api/controllers/well-known.spec.js:24:7)
```

I've tried exploring alternatives with `expect().to.throw` / [chai-as-promised](https://www.npmjs.com/package/chai-as-promised) but it's much less flexible and the syntax gets easily confusing...